Documentation/teaching/labs/arm_kernel_development.rst

```diff
@@ -0,0 +1,330 @@
+=========================
+Kernel Development on ARM
+=========================
+
+Lab objectives
+==============
+
+* get a feeling of what System on a Chip (SoC) means
+* get familiar with embedded world using ARM as support architecture
+* understand what a Board Support Package means (BSP)
+* compile and boot an ARM kernel with Qemu using i.MX6UL platform as an example
+* get familiar with hardware description using Device Trees
+
+System on a Chip
+================
+
+A System on a Chip (**SoC**) is an integrated circuit (**IC**) that integrates an entire system onto it. The components
+that can be usually found on an SoC include a central processing unit (**CPU**), memory, input/output ports, storage devices
+together with more sophisticated modules like audio digital interfaces, neural processing units (**NPU**) or graphical
+processing units (**GPU**).
+
+SoCs can be used in various applications most common are:
+ - consumer electronics (TV sets, mobile phones, video game consoles)
+ - industrial computers (medical imaging, etc)
+ - automotive
+ - home appliances
+
+The leading architecture for SoCs is **ARM**. Worth mentioning here is that there are also x86-based SoCs platforms. Another thing
+we need to keep an eye on is **RISC-V** an open standard instruction set architecture.
+
+A simplified view of an **ARM** platform is shown in the image below:
+
+.. image:: ../res/schematic.png
+   :align: center
+
+We will refer as a reference platform at NXP's `i.MX6UL <imx6ul>`_ platform, but in general all SoC's contain the following building blocks:
+
+  - one or more CPU cores
+  - a system bus
+  - clock and reset module
+
+    - PLL
+    - OSC
+    - reset controller
+
+ - interrupt controller
+ - timers
+ - memory controller
+ - peripheral controllers
+
+   - `I2C <https://en.wikipedia.org/wiki/I%C2%B2C>`_
+   - `SPI <https://en.wikipedia.org/wiki/Serial_Peripheral_Interface>`_
+   - `GPIO <https://en.wikipedia.org/wiki/General-purpose_input/output>`_
+   - `Ethernet <https://en.wikipedia.org/wiki/Network_interface_controller>`_ (for network)
+   - `uSDHC <https://en.wikipedia.org/wiki/MultiMediaCard>`_ (for storage)
+   - USB
+   - `UART <https://en.wikipedia.org/wiki/Universal_asynchronous_receiver-transmitter>`_
+   - `I2S <https://en.wikipedia.org/wiki/I%C2%B2S>`_ (for sound)
+   - eLCDIF (for LCD Panel)
+
+Here is the complete block diagram for i.MX6UL platform:
+
+.. image::  https://www.nxp.com/assets/images/en/block-diagrams/IMX6UL-BD.jpg
```

Comment:
We should probably download these images and store them in the repo, in case the URLs are no longer accessible.

Reply:
I was concerned about copyright. I should ask NXP about permission to download.